src/sonic-dhcp-server/dhcp_server/common/dhcp_db_monitor.py

```diff
@@ -92,40 +144,113 @@ def _check_dhcp_server_update(self, enabled_dhcp_interfaces):
+        self.subscribe_dhcp_server_range_table = swsscommon.SubscriberStateTable(self.db_connector.config_db,
+                                                                                 DHCP_SERVER_IPV4_RANGE)
+        self.subscribe_dhcp_server_options_table = swsscommon.SubscriberStateTable(self.db_connector.config_db,
+                                                                                   DHCP_SERVER_IPV4_CUSTOMIZED_OPTIONS)
+        self.subscribe_vlan_table = swsscommon.SubscriberStateTable(self.db_connector.config_db, VLAN)
+        self.subscribe_vlan_member_table = swsscommon.SubscriberStateTable(self.db_connector.config_db, VLAN_MEMBER)
+        self.subscribe_vlan_intf_table = swsscommon.SubscriberStateTable(self.db_connector.config_db, VLAN_INTERFACE)
+        # Subscribe dhcp_server and vlan related tables.
+        self.sel.addSelectable(self.subscribe_dhcp_server_table)
+        self.sel.addSelectable(self.subscribe_dhcp_server_port_table)
+        self.sel.addSelectable(self.subscribe_dhcp_server_range_table)
+        self.sel.addSelectable(self.subscribe_dhcp_server_options_table)
+        self.sel.addSelectable(self.subscribe_vlan_table)
+        self.sel.addSelectable(self.subscribe_vlan_member_table)
+        self.sel.addSelectable(self.subscribe_vlan_intf_table)
+
+    def _do_check(self, check_param):
+        if "enabled_dhcp_interfaces" not in check_param or "used_range" not in check_param or \
+           "used_options" not in check_param:
+            # Need to regenerate config, so clear table update event to avoid unlimited updates
+            self._pop_db_update_event(self.subscribe_dhcp_server_table)
+            self._pop_db_update_event(self.subscribe_dhcp_server_port_table)
+            self._pop_db_update_event(self.subscribe_dhcp_server_range_table)
+            self._pop_db_update_event(self.subscribe_dhcp_server_options_table)
+            self._pop_db_update_event(self.subscribe_vlan_table)
+            self._pop_db_update_event(self.subscribe_vlan_member_table)
+            self._pop_db_update_event(self.subscribe_vlan_intf_table)
+            syslog.syslog(syslog.LOG_ERR, "Cannot get enough check param")
+            return True
+        enabled_dhcp_interfaces = check_param["enabled_dhcp_interfaces"]
+        used_range = check_param["used_range"]
+        used_options = check_param["used_options"]
+        need_refresh = self._check_dhcp_server_update(enabled_dhcp_interfaces)
+        need_refresh |= self._check_vlan_update(enabled_dhcp_interfaces)
+        need_refresh |= self._check_vlan_intf_update(enabled_dhcp_interfaces)
+        need_refresh |= self._check_dhcp_server_port_update(enabled_dhcp_interfaces)
+        need_refresh |= self._check_dhcp_server_range_update(used_range)
+        need_refresh |= self._check_dhcp_server_option_update(used_options)
+        need_refresh |= self._check_vlan_member_update(enabled_dhcp_interfaces)
+        return need_refresh
```

Comment:
DhcpRelaydDbMonitor returns a (Bool, Bool, Bool) tuple, but here you returns a single Bool.
What is your consideration? The unaligned return type will result in the abstraction is unpredictable

Reply:
Added _DbEventExecutor to handle common db update event and removed basic class